server/model/order.js

```diff
@@ -0,0 +1,33 @@
+const utils = require('../utils/utils');
+
+const ordersList = [];
+
+const getAllOrders = () => (ordersList);
+
+const getOrder = (id) => {
+  if (ordersList.some((e) => e.id === id)) {
+    return ordersList.find((order) => order.id === id);
+  }
+  return false;
+};
+
+const createOrder = (order) => {
+  const id = utils.idGenerator.generateId();
+  ordersList.push({ ...order, id });
+};
+
+const getOrdersByUsername = (username) => {
+  if (ordersList.some((e) => e.userName === username)) {
+    const usersOrders = ordersList.filter((order) => order.userName === username);
+    usersOrders.sort((a, b) => (b.date - a.date));
+    return usersOrders;
+  }
```

Comment:
There is no need for finding orders of user with some function
If you use filter function and there are no items return the empty list.
this is not an error.
if the user doesn't exist in the user list this should be an error